src/syscheckd/fim_diff_changes.c

```diff
@@ -996,17 +968,21 @@ int fim_diff_process_delete_registry(const char *key_name, int arch){
         snprintf(full_path, PATH_MAX, "%s/registry/[x32] %s", DIFF_DIR, encoded_key);
     }
 
-    if(fim_diff_delete_compress_folder(full_path) == -1){
-        return -1;
+    ret = fim_diff_delete_compress_folder(full_path);
+    if(ret == -1){
```

Comment:
Is this case covered by any unit test?

Reply:
Yes, this case is contemplate in the test `test_fim_diff_process_delete_registry_delete_error`